packages/renderer/src/lib/engine/Prune.svelte

```diff
@@ -27,7 +27,7 @@ async function prune(type: string) {
     case 'containers':
       engines.forEach(async engine => {
         try {
-          await window.pruneContainers(engine.id);
+          window.pruneContainers(engine.id);
```

Comment:
doing that we won't catch anything in the catch block, it will throw probably an `Uncaught Exception`

if the reason behind is to make it parallel, you could use map iteration with Promise.all()

I think expectation of the `prune` method is to complete when everything has been pruned.
For example the button 'prune` should be 'in progress/disabled' until the action is not finished.
Here there is no linger a guarantee

Reply:
Yup, this is implemented in all the other buttons / pods / images / volumes removal functions too.

Should we just make them all `await` for now and then add the progress button in a different PR?

Reply:
Updated with adding await @benoitf

Reply:
Ok let's keep await for now
We could replace forEach with await Promise.all(list.map()) but it's like improvements